torchmetrics/regression/explained_variance.py

```diff
@@ -113,11 +113,11 @@ def update(self, preds: Tensor, target: Tensor) -> None:
             target: Ground truth values
         """
         n_obs, sum_error, sum_squared_error, sum_target, sum_squared_target = _explained_variance_update(preds, target)
-        self.n_obs = self.n_obs + n_obs
-        self.sum_error = self.sum_error + sum_error
-        self.sum_squared_error = self.sum_squared_error + sum_squared_error
-        self.sum_target = self.sum_target + sum_target
-        self.sum_squared_target = self.sum_squared_target + sum_squared_target
+        self.n_obs: Tensor = self.n_obs + n_obs
+        self.sum_error: Tensor = self.sum_error + sum_error
+        self.sum_squared_error: Tensor = self.sum_squared_error + sum_squared_error
+        self.sum_target: Tensor = self.sum_target + sum_target
+        self.sum_squared_target: Tensor = self.sum_squared_target + sum_squared_target
```

Comment:
it's better to add the typing already outside (where you usually define static attributes) and not initialize it then typing lately here. 

So something like 

```python

class MyMetric:
    a: torch.Tensor
    def __init__(self):
        super().__init__()
        self.add_state('a', default=torch.tensor(0.))
```